db_test.go

```diff
@@ -1098,3 +1100,51 @@ func TestOverlappingBlocksDetectsAllOverlaps(t *testing.T) {
 		{Min: 8, Max: 9}: {nc1[8], nc1[9]},                                 // 7-10, 8-9
 	}, OverlappingBlocks(nc1))
 }
+
+func TestChunkAtBlockBoundary(t *testing.T) {
```

Comment:
Please add comment that this is regression test for issue https://github.com/prometheus/tsdb/issues/347